ADR for the output port lifecycle (#2225)
conveyordata/data-product-portal: docs/adr/0010-output-port-lifecycle.md

# Output Port Lifecycle Management

## Context and Problem Statement

Data Producers want to have the ability to control the visibility of their Output Ports during development.
Data Consumers want to have visibility on the state of a Data Output, in order to estimate if it is a good fit.

## Decision Drivers

* Ease of use
* Control for the Data Producer
* Clarity for the Data Consumer

## Considered Options

> * **Option 1: Simple toggle** A data output can be switched between 'draft' and 'published'.

Reviewer: ```suggestion
* **Option 1: Simple toggle** An output port can be switched between 'draft' and 'published'.
```

Maybe best to change this everywhere

Reviewer 2: changed it everywhere.

  Draft data outputs are not visible to people outside the data product, published outputs are visible.

> * **Option 2: State machine** A data output has four states: 'draft', 'active', 'deprecated' and 'retired'.

Reviewer: This can be build on the current lifecycles already added to output ports, which are configurable.
It could just be a enum on the configurable states, so customers can decide certain states are invisble, visible or deprecated (the version where for current access it's still visible, but not for new access requests).

Reviewer: You added a fourth view option in your proposal, I like it
